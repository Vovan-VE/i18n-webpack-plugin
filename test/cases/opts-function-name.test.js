import fs from 'fs';
import processFile from '../cases.setup';

describe('options.functionName', () => {
  describe('the object notation', () => {
    let extracted;

    beforeAll(() => {
      const options = {
        functionName: '__',
      };

      return processFile('opts-function-name.code.js', options)
        .then(({ file }) => {
          extracted = fs.readFileSync(file, 'utf8');
        });
    });

    it('should return extracted keys', () => {
      expect(extracted).toMatchSnapshot();
    });
  });
});
