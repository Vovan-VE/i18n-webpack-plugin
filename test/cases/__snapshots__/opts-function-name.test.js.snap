// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options.functionName the object notation should return extracted keys 1`] = `
"{
  \\"category\\": {
    \\"key\\": \\"\\"
  }
}"
`;
